src/earn/EarnEnterAmount.tsx

```diff
@@ -421,55 +349,31 @@ function EarnEnterAmount({ route }: Props) {
           </Text>
-          <View style={styles.inputBox}>
-            <View style={styles.inputRow}>
-              <AmountInput
-                inputRef={tokenAmountInputRef}
-                inputValue={tokenAmountInput}
-                onInputChange={onTokenAmountInputChange}
-                inputStyle={styles.inputText}
-                autoFocus
-                placeholder={new BigNumber(0).toFormat(2)}
-                testID="EarnEnterAmount/TokenAmountInput"
-              />
-              <Touchable
-                borderRadius={TOKEN_SELECTOR_BORDER_RADIUS}
-                onPress={onTokenPickerSelect}
-                style={styles.tokenSelectButton}
-                disabled={!dropdownEnabled}
-                testID="EarnEnterAmount/TokenSelect"
-              >
-                <>
-                  <TokenIcon token={inputToken} size={IconSize.SMALL} />
-                  <Text style={styles.tokenName}>{inputToken.symbol}</Text>
-                  {dropdownEnabled && <DownArrowIcon color={Colors.gray5} />}
-                </>
-              </Touchable>
-            </View>
-            <View style={styles.localAmountRow}>
-              <AmountInput
-                inputValue={transactionToken.priceUsd ? localAmountInput : '-'}
-                onInputChange={onLocalAmountInputChange}
-                inputRef={localAmountInputRef}
-                inputStyle={styles.localAmount}
-                placeholder={`${localCurrencySymbol}${new BigNumber(0).toFormat(2)}`}
-                testID="EarnEnterAmount/LocalAmountInput"
-                editable={!!transactionToken.priceUsd}
-              />
-            </View>
-          </View>
-          {tokenAmount && prepareTransactionsResult && !isWithdrawal && (
+          <TokenEnterAmount
+            autoFocus
+            testID="EarnEnterAmount"
+            token={inputToken}
+            inputValue={amount}
+            inputRef={inputRef}
+            tokenAmount={processedAmounts.token.displayAmount}
+            localAmount={processedAmounts.local.displayAmount}
+            onInputChange={handleAmountInputChange}
+            amountType={amountType}
+            toggleAmountType={handleToggleAmountType}
+            onOpenTokenPicker={dropdownEnabled ? onOpenTokenPicker : undefined}
+          />
+          {processedAmounts.token.bignum && prepareTransactionsResult && !isWithdrawal && (
             <TransactionDepositDetails
               pool={pool}
               token={inputToken}
-              tokenAmount={tokenAmount}
+              tokenAmount={processedAmounts.token.bignum}
               prepareTransactionsResult={prepareTransactionsResult}
               feeDetailsBottomSheetRef={feeDetailsBottomSheetRef}
               swapDetailsBottomSheetRef={swapDetailsBottomSheetRef}
               swapTransaction={swapTransaction}
             />
           )}
-          {tokenAmount && isWithdrawal && (
+          {isWithdrawal && (
```

Comment:
It seems like there is no need to check for tokenAmount at all as we need to show withdrawal details whenever we go through the withdrawal flow.

Reply:
i think the removal of this check is what is causing the empty box to be displayed before the fees are ready

Reply:
Hmm, I'll need to adjust it then. Previously when there was "available" section - it was showing at least one static block that was showing up without conditions. Now as it was removed, there are two conditional blocks: claiming rewards and fees. When amount is entered - the `tokenAmount` will be not empty and `isWithdrawal` will be true but both conditions will still evaluate to "false" for some time so the empty box will be shown anyway. 

I'll adjust this so it would only show up if at least one condition passes.